cmd/keyaddr/handlers.go

```diff
@@ -72,6 +74,136 @@ func newKey(this js.Value, args []js.Value) interface{} {
 	return nil
 }
 
+// JS Usage: newEdKey(cb) --> no args
+func newEdKey(this js.Value, args []js.Value) interface{} {
+	go func(args []js.Value) {
+		logDebug("newEdKey")
+
+		// clean args
+		callback, _, err := handleArgs(args, 0, "newEdKey")
+		if err != nil {
+			return
+		}
+
+		// do work
+		public, private, err := signature.Generate(signature.Ed25519, nil)
+		if err != nil {
+			jsLogReject(callback, "error creating new ed key: %s", err)
+			return
+		}
+
+		pubkeyText, err := public.MarshalText()
+		if err != nil {
+			jsLogReject(callback, "error marshalling new public ed key: %s", err)
+			return
+		}
+
+		privkeyText, err := private.MarshalText()
+		if err != nil {
+			jsLogReject(callback, "error marshalling new private ed key: %s", err)
+			return
+		}
+
+		// Make an map[string]interface{} so syscall will turn it into a js object.
+		obj := make(map[string]interface{})
+		obj["pubkey"] = string(pubkeyText)
+		obj["privkey"] = string(privkeyText)
+
+		// return result
+		callback.Invoke(nil, js.ValueOf(obj))
+		return
+	}(args)
+	return nil
+}
+
+// JS Usage: newEdKey(seed, cb)
+func newEdKeyFromSeed(this js.Value, args []js.Value) interface{} {
+	go func(args []js.Value) {
+		logDebug("newEdKeyFromSeed")
+
+		// clean args
+		callback, remainder, err := handleArgs(args, 1, "newEdKeyFromSeed")
+		if err != nil {
+			return
+		}
+
+		seedString := remainder[0].String()
+		seed, err := hex.DecodeString(seedString)
+		if err != nil {
+			jsLogReject(callback, "error decoding data from hex: %s", err)
+			return
+		}
+
```

Comment:
Should we check for correct seed length here? I don't think it's necessary, since a bad key length will just cause `GenerateFromSeed` to return an error.